src/toolchain_override.rs

```diff
@@ -1,18 +1,21 @@
 use anyhow::{bail, Result};
 use semver::Version;
-use serde::{Deserialize, Serialize};
+use serde::de::Error;
+use serde::ser::SerializeStruct;
+use serde::{Deserialize, Deserializer, Serialize};
+use std::fmt;
 use std::str::FromStr;
 use std::{collections::HashMap, path::PathBuf};
+use time::Date;
 use toml_edit::{de, ser, value, Document};
 use tracing::{info, warn};
 
+use crate::channel::{is_beta_toolchain, LATEST, NIGHTLY};
+use crate::constants::DATE_FORMAT;
+use crate::toolchain::DistToolchainDescription;
 use crate::{
-    constants::FUEL_TOOLCHAIN_TOML_FILE,
-    download::DownloadCfg,
-    file,
-    path::get_fuel_toolchain_toml,
-    target_triple::TargetTriple,
-    toolchain::{DistToolchainName, Toolchain},
+    constants::FUEL_TOOLCHAIN_TOML_FILE, download::DownloadCfg, file,
```

Comment:
Above we have `use crate::constants::DATE_FORMAT`, but then we also import from `constants` in this import line? We should pick one approach and try to be more consistent with the way we import items. Not a big deal - maybe something `cargo fmt` can do for us automatically in the future.

Reply:
Thanks for the catch! Cursed auto-complete doesn't seem to be able to reconcile nested imports with unnested ones.

Does `cargo fmt` catch this? Didn't seem to work for me.

Reply:
Ahh I meant that hopefully in the future cargo fmt would be able to catch this for us - I'm not sure if it can now, but I don't think it does by default at least.